src/main/user-impl/java/com/mysql/cj/jdbc/ha/ConnectionProxy.java

```diff
@@ -177,12 +182,14 @@ public synchronized Object invoke(Object proxy, Method method, Object[] args)
       return executeMethodDirectly(methodName, args);
     }
 
+    Object[] argsCopy = args == null ?  null : Arrays.copyOf(args, args.length);
```

Comment:
Don't think we need this

Reply:
As discussed. Let's keep it to avoid side effects.